In [1]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency


In [14]:
df=pd.read_csv("ab.csv")
df

version  not-converted  converted
0       A          18000      32000
1       B          20000      30000

In [15]:
df = df.set_index('version')
df

not-converted  converted
version                          
A                18000      32000
B                20000      30000

In [17]:
# Transform a Pandas dataframe into numpy array
observed = df.values
observed

array([[18000, 32000],
       [20000, 30000]])

In [19]:
result = chi2_contingency(observed)
result

(169.60955008488963, 9.0042014755951057e-39, 1, array([[ 19000.,  31000.],
        [ 19000.,  31000.]]))

#### Give the results of (169.60955008488963, 9.0042014755951057e-39, 1, array([[ 19000.,  31000.],[ 19000.,  31000.]]))
### (Chi2, p-value,degree of freedom, The expected frequencies, based on the marginal sums of the table.)

In [23]:
chisq, p = result[:2]

print ('chisq = {}, p-value = {}'.format(chisq, p))

chisq = 169.60955008488963, p-value = 9.004201475595106e-39


### Here the p-value is smaller than 0.05, suggesting there is a significant difference between app version A and B.  Here the conversion rate of A is better than B, so we might go for version A.  